File: node-express/src/envHelpers.ts
export function setDefaultEnvVar(
  key: string,
  defaultValue: string | undefined
) {
  const value = process.env[key];
  if (value === undefined) {
    process.env[key] = defaultValue;
  }
}

export function getRequiredEnvVar(key: string): string {
  const value = process.env[key];
  if (value === undefined) {
    throw Error(`${key} env var does not exist!`);
  }
  return value;
}

File: rust-actix/src/env_helpers.rs
use std::{env, str::FromStr};

pub fn set_default_env_var(key: &str, value: &str) {
    if env::var(key).is_err() {
        env::set_var(key, value);
    }
}

pub fn cast_required_env_var<F: FromStr>(key: &str) -> F
where
    <F as FromStr>::Err: std::fmt::Debug,
{
    env::var(key)
        .expect(&format!("{} env var does not exist!", key))
        .parse::<F>()
        .expect(&format!("Error casting {} env var!", key))
}
